SubsoniciousKit/SubsoniciousKit/Network/Response/SubsonicResponse.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  SubsonicResponse.swift
+//  SubsoniciousKit
+//
+//  Created by Bilal on 05/07/2020.
+//  Copyright © 2020 Bilal Benlarbi. All rights reserved.
+//
+
+import Foundation
+
+public struct SubsonicResponse: Decodable {
+    public let status: Status
+    public let error: Error?
+
+    enum CodingKeys: String, CodingKey {
+        case responseContainer = "subsonic-response"
+    }
+
+    enum ResponseCodingKeys: String, CodingKey {
+        case status
+        case error
+    }
+
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+
+        let responseContainer = try container.nestedContainer(keyedBy: ResponseCodingKeys.self, forKey: .responseContainer)
+        status = try responseContainer.decode(Status.self, forKey: .status)
+        error = try responseContainer.decodeIfPresent(Error.self, forKey: .error)
```

Comment:
if you have an error, you should throw it, instead of keep it here inside your struct...

Reply:
This could change a lot of things, but in a better way I think :)

Reply:
good idea :) I've tried something, let me know if you're fine with it

Reply:
I think that's fine :)